airbyte-config/init/src/main/java/io/airbyte/config/init/ApplyDefinitionsHelper.java

```diff
@@ -20,26 +22,21 @@
  * Helper class used to apply actor definitions from a DefinitionsProvider to the database. This is
  * here to enable easy reuse of definition application logic in bootloader and cron.
  */
+@Singleton
+@Requires(bean = ConfigRepository.class)
+@Requires(bean = JobPersistence.class)
 @Slf4j
 public class ApplyDefinitionsHelper {
 
   private final ConfigRepository configRepository;
-  private final DefinitionsProvider definitionsProvider;
+  private final Optional<DefinitionsProvider> definitionsProviderOptional;
   private final JobPersistence jobPersistence;
 
-  // Remove once cloud has been migrated
-  @Deprecated(forRemoval = true)
-  public ApplyDefinitionsHelper(final ConfigRepository configRepository, final DefinitionsProvider definitionsProvider) {
-    this.configRepository = configRepository;
-    this.definitionsProvider = definitionsProvider;
-    this.jobPersistence = null;
-  }
-
   public ApplyDefinitionsHelper(final ConfigRepository configRepository,
-                                final DefinitionsProvider definitionsProvider,
+                                final Optional<DefinitionsProvider> definitionsProviderOptional,
```

Comment:
Just saw this afterwards, why do we have this `@Requires` and `Optional` distinction here? Shouldn't all three arguments always be required?

Reply:
@gosusnp Based on the existing code, there is a case where no definition provider is configured.  I'm not sure why that is the case, but for this first pass, I am just trying to maintain the existing functionality.